holonote/annotate/plotting.py

```diff
@@ -0,0 +1,505 @@
+
+        if indicators:
+            layers.append(self.indicators())
+        if editor:
+            layers.append(self.region_editor())
+        return hv.Overlay(layers).collate()
+
+    @property
+    def static_indicators(self):
+        data = self.annotator.get_dataframe(dims=self.kdims)
+        fields_labels = self.annotator.all_fields
+        region_labels = [k for k in data.columns if k not in fields_labels]
+
+        indicator_kwargs = {
+            "data": data,
+            "region_labels": region_labels,
+            "fields_labels": fields_labels,
+            "invert_axes": False,  # Not yet handled
+            "groupby": self.annotator.groupby,
+        }
+
+        if self.region_types == "range":
+            indicator = Indicator.ranges_1d(**indicator_kwargs)
+        elif self.region_types == "range-range":
+            indicator = Indicator.ranges_2d(**indicator_kwargs)
+        elif self.region_types == "point":
+            indicator = Indicator.points_1d(**indicator_kwargs)
+        elif self.region_types == "point-point":
+            indicator = Indicator.points_2d(**indicator_kwargs)
+        else:
+            msg = f"{self.region_types} not implemented"
+            raise NotImplementedError(msg)
+
+        if self.annotator.groupby and self.annotator.visible:
+            indicator = indicator.get(self.annotator.visible)
+            if indicator is None:
+                vis = "', '".join(self.annotator.visible)
+                msg = f"Visible dimensions {vis!r} not in spec"
+                raise ValueError(msg)
+
+        # Set styling on annotations indicator
+        highlight = self.style.indicator_selection
+        highlighters = {opt: self.selected_dim_expr(v[0], v[1]) for opt, v in highlight.items()}
+        indicator = indicator.opts(*self.style.indicator(**highlighters))
+
+        return indicator.overlay() if self.annotator.groupby else hv.NdOverlay({0: indicator})
+
+    def selected_dim_expr(self, selected_value, non_selected_value):
+        self._selected_values.append(selected_value)
+        self._selected_options.append({i: selected_value for i in self.annotator.selected_indices})
+        index_name = (
+            "id"
+            if self.annotator.annotation_table._field_df.index.name is None
+            else self.annotator.annotation_table._field_df.index.name
+        )
+        return hv.dim(index_name).categorize(
+            self._selected_options[-1], default=non_selected_value
+        )
+
+    @property
+    def dim_expr(self):
+        return self._selection_info["dim_expr"]
+
+    def show_region(self):
+        kdims = list(self.kdims)
+        region = {k: v for k, v in self.annotator._region.items() if k in self.kdims}
+
+        if not region:
+            return
+
+        if self.region_types == "range":
+            value = region[kdims[0]]
+            bounds = (value[0], 0, value[1], 1)
+        elif self.region_types == "range-range":
```

Comment:
What is a "range-range" region type? There should only be 'range' whether it is 1D, 2D etc. This isn't declared private so this must be user-level API - imho this should be inferred from the dimensionality of the dimension specification.

Reply:
I also note that `range-range` is not in the list of allowed default_region_type values

Reply:
`AnnotationDisplay` is private in my mind. It is not expected for a user to interact with this class. I can make it private and rename it to `_AnnotationDisplay`. 

The `AnnotationDisplay.regions_types` has nothing to do with `Annotator.default_region_type`. There can be multiple `AnnotationDisplay`s for a single `Annotation`; each of these can have different `region_types` based on the spec.

`AnnotationDisplay.regions_types` is a way to keep up with the dimensionality, which is inferred based on the number of dimensions and their respective region.  

![image](https://github.com/holoviz/holonote/assets/19758978/332d404c-662d-42a7-84bd-e7c6ff56dc47)

Reply:
I think it is confusing to have two variables with the same name meaning almost-but-not-quite the same thing in different contexts. My point is that a new semantic name is needed....e.g. `display_region_type` maybe? 

What does `range-range` represent? That there are two dimensions, each which are ranges (so probably mapping to a rectangle)?

Reply:
> display_region_type maybe?

Will rename it to that. 

> What does range-range represent? That there are two dimensions, each which are ranges (so probably mapping to a rectangle)?

Correct :+1:

Reply:
renamed to `region_format`

Reply:
Resolved in [b7b6dc4](https://github.com/holoviz/holonote/pull/51/commits/b7b6dc410e834e8a64b3e7553371b80e13de3381)